bin/dothething

```diff
@@ -30,147 +34,247 @@ rescue OptionParser::InvalidOption => e
  exit 1
 end.parse!
 
-class State
- #Loads the config file
- def initialize(state)
-  @statefile = state 
-  @sf = YAML.load_file(@statefile)
- end
+#If no command was entered display help and exit 
+if options.empty?
+ puts opt_parser
+ exit 1
+end
 
- #Toggles value in state hash
- def toggle(name)
-  #Finds all true/false values in hash
-  t = @sf[name].key(true)
-  f = @sf[name].key(false) 
+module ConfigFile
+ class Config
+  #Load the config file
+  def initialize
+   configfile = File.join($path, 'config.yml')
+   @config = YAML.load_file(configfile)
+  end
+  
+  #Check if a setting has been set
+  def setting_exists?(setting)
+   if @config[setting] != nil then
+    return true 
+   else
+    puts "The config file does not contain a path for the setting: #{setting}"
+    exit 1 
+   end 
+  end
+  
+  #Set the directory for the job   
+  def jobdir(name)
+   setting_exists?('jobdir')
+   @jobdir = @config['jobdir']
+   
+   #Check if the path given in the file is absolute or relative   
+   if @jobdir.include? "../"
+    @jobdir = File.join($path, @jobdir, name)
+   else
+    @jobdir = File.join(@jobdir, name)
+   end      
+  end
+  
+  #Check if a file exists
+  def file_exist?(setting)
```

Comment:
I'd consider accepting the actual path to a file in this method rather than the setting, its no extra trouble to lookup config value before calling the method and this method would then only concern itself with checking if a file exists. I'd also consider making it a private method.

Reply:
The file_exist? method is now private and takes the file path as a parameter